buildSrc/src/main/groovy/com/uber/okbuck/generator/BuckFileGenerator.groovy

```diff
@@ -97,7 +94,7 @@ final class BuckFileGenerator {
                     }
                     break
                 default:
-                    break
+                    throw new IllegalArgumentException("ProjectType not handled: $projectType")
```

Comment:
An exception is good only if we can offer some mitigation. When a project type is not handled, we should put a message in the exception to excldue the project using `okbuck.buckProjects` extension. Otherwise it is not actionable.

I am ok taking the exception, if we make the error message clearer